app/src/main/java/com/sdp13epfl2021/projmag/model/SectionBaseManager.kt

```diff
@@ -0,0 +1,25 @@
+package com.sdp13epfl2021.projmag.model
+
+class SectionBaseManager {
+    companion object{
+        private val sectionsList = listOf<String>(
+            "Chemistry", "Mathematics", "Physics", "Computer Science", "Communication Systems",
+            "Electrical engineering", "Mechanical Engineering", "MicroEngineering", "Life sciences",
+            "Architecture", "Civil Engineering", "Environmental Sciences and Engineering"
+        )
+    }
+
+    fun sectionList(): List<String> {
+        return sectionsList
+    }
+
+    fun isListValid(listToBeChecked : List<String>) : Boolean{
+        for(section in listToBeChecked){
+            if(!sectionsList.contains(section)){
+                return false
+            }
+        }
+        return true
+
```

Comment:
Could be simplified by this
```suggestion
        listToBeChecked.all(sectionsList::contains)
```

Reply:
Begone, odersky fan, we do it the imperative way like it has been done for thousands of years before us

Reply:
**!**(I aggree with you)

Reply:
!(I aggree with you)